### PR TITLE
Cairo 0.10.3

### LINKED ISSUE #1222: Cairo-lang 0.10.3 integration
https://github.com/starkware-libs/cairo-lang/releases/tag/v0.10.3-pre

### PR DESCRIPTION
**Pull request by kasperski95**
Closes #1222

### CODE REVIEW

File: tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo [hunk 1]
```diff
@@ -246,9 +246,7 @@ func test_constructor_no_args_executed{syscall_ptr: felt*, range_check_ptr}() {
     alloc_locals;
```
Comment (MaksymilianDemitraszek) [inline tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo:246]: Are those changes related to the PR?

File: pyproject.toml [hunk 2]
```diff
@@ -88,7 +89,7 @@ update_cli_docs = "python ./scripts/generate_reference_docs.py"
 shell = "pytest -n auto tests/e2e"
 
 [tool.poe.tasks.test_integration]
-shell = "pytest -n auto scripts tests/integration"
+shell = "pytest -n auto --random-order scripts tests/integration"
```
Comment (MaksymilianDemitraszek) [inline pyproject.toml:92]: It is not related to the PR

File: protostar/starknet/forkable_starknet.py [hunk 3]
```diff
@@ -40,7 +40,8 @@ def copy_and_adapt_contract(self, deployed_contract: StarknetContract):
         )
 
     def fork(self):
-        return ForkableStarknet(state=self.cheatable_state.copy())
+        state_copy = self.cheatable_state.copy()
+        return ForkableStarknet(state=state_copy)
```
Comment (MaksymilianDemitraszek) [inline protostar/starknet/forkable_starknet.py:44]: This change is not related to the PR

**Review by MaksymilianDemitraszek — APPROVED** [review]
Please remove unrelated changes from the PR

File: pyproject.toml (hunk 2 above)
Comment (kasperski95) [reply pyproject.toml:92]: It is. WIthout it, the problem with devnet/CI occurs. The diff should disappear after rebasing.

File: tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo (hunk 1 above)
Comment (kasperski95) [reply tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo]: Reversed changes.

File: pyproject.toml (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply pyproject.toml]: It should have been fixed in a separate PR and then rebased
